JPA overridden properties from MapperSuperclass is not evaluated correctly

### Describe the bug

I am working on a spring-boot to quarkus migration project and I came through this issue when copying my JPA entities as it is from spring boot to quarkus. I have replicated the issue in a sample project (https://github.com/faskan/quarkus-hibernate-sample)

There exists a MappedSuperClass with name AbstractEntity, with the following field which is mapped to database column `name`
```
@MappedSuperclass
public class AbstractEntity {

    @Column(length = 40, unique = true)
    private String name;

}
```

And my entity class overrides the name property like shown below. In my spring boot project, the overridden property (name) gets evaluated correctly. But in quarkus, it gets evaluated to null.

```
@Entity
@Table(name = "known_fruits")
@NamedQuery(name = "Fruits.findAll", query = "SELECT f FROM Fruit f ORDER BY f.name", hints = @QueryHint(name = "org.hibernate.cacheable", value = "true"))
@Cacheable
public class Fruit extends AbstractEntity {

    @Id
    @SequenceGenerator(name = "fruitsSequence", sequenceName = "known_fruits_id_seq", allocationSize = 1, initialValue = 10)
    @GeneratedValue(generator = "fruitsSequence")
    private Integer id;

    @Column(length = 40, unique = true)
    private String name;

    public Fruit() {
    }

    public Fruit(String name) {
        this.name = name;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

}
```

The test case results in failure
```
[ERROR] Failures: 
[ERROR]   FruitsEndpointTest.testListAllFruits:21 3 expectations failed.
Response body doesn't match expectation.
Expected: a string containing "Cherry"
  Actual: [{"name":null,"id":2},{"name":null,"id":3},{"name":null,"id":1}]

Response body doesn't match expectation.
Expected: a string containing "Apple"
  Actual: [{"name":null,"id":2},{"name":null,"id":3},{"name":null,"id":1}]

Response body doesn't match expectation.
Expected: a string containing "Banana"
  Actual: [{"name":null,"id":2},{"name":null,"id":3},{"name":null,"id":1}]
```

The test case gets successful when I remove `@MappedSuperclass` annotation, but I am not sure if it would break something else.

I expect hibernate orm to evaluate the overridden property correctly when the MappedSuperclass annotation is present.

### Expected behavior

I expect hibernate orm to evaluate the overridden property correctly when the MappedSuperclass annotation is present.

### Actual behavior

The value of the overridden properties gets evaluated to null when the MappedSuperclass annotation is present in the super class.

### How to Reproduce?

Steps to reproduct:

1. Clone this repo https://github.com/faskan/quarkus-hibernate-sample
2. run `mvn quarkus:dev`
3. Test cases will fail because of incorrect evaluation of properties

### Output of `uname -a` or `ver`

Darwin Faisals-MacBook-Pro.local 21.6.0 Darwin Kernel Version 21.6.0: Wed Aug 10 14:25:27 PDT 2022; root:xnu-8020.141.5~2/RELEASE_X86_64 x86_64

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

2.12.1.Final

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -88,7 +88,7 @@
         <commons-lang3.version>3.12.0</commons-lang3.version>
         <commons-codec.version>1.15</commons-codec.version>
         <classmate.version>1.5.1</classmate.version>
-        <hibernate-orm.version>5.6.11.Final</hibernate-orm.version> <!-- When updating, align bytebuddy.version to Hibernate needs as well (just below): -->
+        <hibernate-orm.version>5.6.12.Final</hibernate-orm.version> <!-- When updating, align bytebuddy.version to Hibernate needs as well (just below): -->
         <bytebuddy.version>1.12.9</bytebuddy.version> <!-- Version controlled by Hibernate ORM's needs -->
         <hibernate-reactive.version>1.1.7.Final</hibernate-reactive.version>
         <hibernate-validator.version>6.2.5.Final</hibernate-validator.version>

--- a/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/GraalVMFeatures.java
+++ b/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/GraalVMFeatures.java
@@ -1,0 +1,24 @@
+package io.quarkus.hibernate.orm.deployment;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.BuildSteps;
+import io.quarkus.deployment.builditem.NativeImageFeatureBuildItem;
+
+/**
+ * Activates the native-image features included in the module
+ * org.hibernate:hibernate-graalvm.
+ */
+@BuildSteps
+public class GraalVMFeatures {
+
+    @BuildStep
+    NativeImageFeatureBuildItem staticNativeImageFeature() {
+        return new NativeImageFeatureBuildItem("org.hibernate.graalvm.internal.GraalVMStaticFeature");
+    }
+
+    @BuildStep
+    NativeImageFeatureBuildItem queryParsingSupportFeature() {
+        return new NativeImageFeatureBuildItem("org.hibernate.graalvm.internal.QueryParsingSupport");
+    }
+
+}

